Bottom: 67375734b46beab50d73c975ee8a8a58ec82fd67
Top:    7129e1d401e4a4e8d24b107f810b821e9c2d5ae9
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-13 21:42:26 +0000

Refresh of random-pile-of-improvements

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 81d8574..87a622e 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1115,53 +1115,20 @@ rounded_contour_bounds_touch (const BoxType * b, void *cl)
   rtree_over = pb;
 
   av = &looping_over->head;
-  do				/* Loop over the edges in the smaller contour */
+  do  /* Loop over the edges in the a contour */
     {
       /* check this vertex for any insertions */
       info.v = av;
-#if 0
-      /* check this edge for any insertions */
-      double dx;
-      /* compute the slant for region trimming */
-      dx = EDGE_FORWARD_VERTEX (av)->point[0] - EDGE_BACKWARD_VERTEX (av)->point[0];
-      if (dx == 0)
-	info.m = 0;
-      else
-	{
-	  info.m = (EDGE_FORWARD_VERTEX (av)->point[1] - EDGE_BACKWARD_VERTEX (av)->point[1]) / dx;
-	  info.b = EDGE_BACKWARD_VERTEX (av)->point[1] - info.m * EDGE_BACKWARD_VERTEX (av)->point[0];
-	}
-      box.X2 = (box.X1 = EDGE_BACKWARD_VERTEX (av)->point[0]) + 1;
-      box.Y2 = (box.Y1 = EDGE_BACKWARD_VERTEX (av)->point[1]) + 1;
-#endif
 
       box.X2 = (box.X1 = av->point[0] - 1) + 3; /* NB: We expand the search box to ensure we catch edges which may round to this coordinate */
       box.Y2 = (box.Y1 = av->point[1] - 1) + 3;
 
-#if 0
-      /* fill in the segment in info corresponding to this node */
-      if (setjmp (info.sego) == 0)
-	{
-	  info.debug = false;
-	  r_search (looping_over->tree, &box, NULL, get_seg, &info);
-	  g_error ("Did not find segment in contour tree!");
-	}
-
-      /* If we're going to have another pass anyway, skip this */
-      if (info.s->intersected && info.node_insert_list != NULL)
-	continue;
-#endif
-
       if (setjmp (restart))
 	continue;
 
       /* NB: If this actually hits anything, we are teleported back to the beginning */
-//      info.tree = rtree_over->tree;
-//      if (info.tree)
-//	if (UNLIKELY (r_search (info.tree, &box /*&info.s->box*/,
       if (rtree_over->tree)
-	if (UNLIKELY (r_search (rtree_over->tree, &box /*&info.s->box*/,
-				NULL/*seg_in_region*/, vertex_in_seg_rounded, &info)))
+	if (UNLIKELY (r_search (rtree_over->tree, &box, NULL, vertex_in_seg_rounded, &info)))
 	  assert (0); /* XXX: Memory allocation failure */
     }
   while ((av = NEXT_VERTEX (av)) != &looping_over->head);
